src/domain/safe/entities/schemas/__tests__/creation-transaction.schema.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { creationTransactionBuilder } from '@/domain/safe/entities/__tests__/creation-transaction.builder';
+import { CreationTransactionSchema } from '@/domain/safe/entities/schemas/creation-transaction.schema';
+import { faker } from '@faker-js/faker';
+import { getAddress } from 'viem';
+
+describe('CreationTransactionSchema', () => {
+  it('should validate a valid creation transaction', () => {
+    const creationTransaction = creationTransactionBuilder().build();
+
+    const result = CreationTransactionSchema.safeParse(creationTransaction);
+
+    expect(result.success).toBe(true);
+  });
+
+  it('should coerce the created date to a Date', () => {
+    const creationTransaction = creationTransactionBuilder().build();
+
+    const result = CreationTransactionSchema.safeParse(creationTransaction);
+
+    expect(result.success && result.data.created).toBeInstanceOf(Date);
+  });
```

Comment:
This is testing faker's ability to generate dates (and not that an ISO date would be coerced into a date). Instead you can do the following:

```typescript
const creationTransaction = {
  ...creationTransactionBuilder().build(),
  created: faker.date.recent().toISOString(),
};
```

Reply:
I improved it in b216217e2edc6f1b0c9409f06db97ee0740e0550, as well as checking the value of the date after coercion.